File: GitToolBox/src/main/java/zielu/gittoolbox/ui/DatePresenter.java
package zielu.gittoolbox.ui;

import java.time.ZonedDateTime;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;
import zielu.gittoolbox.config.DateType;
import zielu.gittoolbox.util.AppUtil;

public interface DatePresenter {
  static DatePresenter getInstance() {
    return AppUtil.getServiceInstance(DatePresenter.class);
  }

  @Nullable
  String format(@NotNull DateType type, @Nullable ZonedDateTime date);
}

File: GitToolBox/src/main/kotlin/zielu/gittoolbox/ui/DatePresenter.kt
package zielu.gittoolbox.ui

import zielu.gittoolbox.config.DateType
import zielu.gittoolbox.util.AppUtil
import java.time.ZonedDateTime

internal interface DatePresenter {

  fun format(type: DateType, date: ZonedDateTime): String

  companion object {
    @JvmStatic
    fun getInstance() = AppUtil.getServiceInstance(DatePresenter::class.java)
  }
}
